app_backend/balance.py

```diff
@@ -0,0 +1,60 @@
+from user import User
+
+
+class BalanceTracker(User):
+    def __init__(self, name: str, dooley_dollars: float, meal_swipes: int):
+        super().__init__(name, dooley_dollars, meal_swipes)
+        """
+
+        name: str
+            The name of the user
+
+        dooley_dollars: float
+            The current balance of dooley dollars
+
+        meal_swipes: int
+            The current amount of remaining meal swipes
+
+        """
+
+    # ------------ Decrementing Balances (for spending) -------------
+
+    def dooley_sub(self, spent: float):
+        """
+        spent: The amount of dooley dollars spent
+        return: The updated Balance Tracker object
+        """
+        if (abs(spent) <= self.dooley_dollars):
+            self.dooley_dollars = self.dooley_dollars - abs(spent)
+        elif (abs(spent) > self.dooley_dollars): # else if
+            self.dooley_dollars = 0
```

Comment:
For dooley, you allow the user to go negative and reset the value to 0. For mealswipes, we can't go negative. 

Instead of being ambigious about functionality, have the function return a boolean where `True` means the function did the operation successfully and `False` means the operation could not be performed. This will let the front end do error handling and show appropriate error messages

Reply:
@RafaelPiloto10 ok

Reply:
@MimiOlayeye, please resolve any comments that you fixed/implemented